### DISCUSSION TITLE
Bundle size - vendor bundle related

### DISCUSSION THREAD

**Discussion by bhagatali** [storybookjs/storybook#11747] [Help] [answered]
Hello! Thanks for all the great work being put into making this an amazing tool for the community to use.💯

I am trying to serve up a very exhaustive storybook we use at work via an `AWS Lambda` fronted by an ALB. Running into problems with the large `vendor` bundle sizes.

To understand how storybook bundling works, I picked up a very sample storybook example from [here](https://storybook.js.org/docs/guides/guide-react/). I wanted to see the bundle output for a very basic storybook with just one sample story. The output from the `build-storybook` without any custom `webpack` configuration produces 2 different vendor bundles. **One of the vendor bundle seems to be 2.5MB in size** and the other one is much smaller (see below). What is the vendor bundle that is 2.5MB? I tried to apply a bundle analyzer and webpacks `splitChunk` but it does not seem to do anything on that larger vendor bundle. Any pointers?

![Screen Shot 2020-07-31 at 1 20 15 PM](https://user-images.githubusercontent.com/6946916/89060185-95014200-d330-11ea-8500-646b188f1c4e.png)

**Comment by bhagatali** [comment]
Ok, so on further digging I realize the 2.5MB of vendor file is coming from the `manager-webpack.config.js`. I still need to find a way to see if this file can be chunked into smaller sizes. Not sure if it is possible. If anyone has any thoughts please let me know.

**Reply by mattrossman** [reply]
Did you end up finding a way to shrink this? The default storybook bundle is insanely huge for what it provides.

**Reply by bhagatali** [reply]
@mattrossman I was able to shrink it using the comments here #4995. I think it is an easy contribution as well. Just not able to get around to submitting the PR. Let me know if you need an example I can send you a gist of the config I used.

**Reply by mattrossman** [reply]
The comments in that thread seem to just describe how to enable a custom webpack config, but what are the actual config options you plugged in to shrink things?

**Reply by bhagatali** [reply]
This is what I plugged in.

```javascript
  config.optimization = {
    ...config.optimization,
    splitChunks: {
      chunks: "all",
      maxInitialRequests: Infinity,
      minSize: 0,
      cacheGroups: {
        vendorPreview: {
          test: /[\\/]node_modules[\\/]/
        }
      },
      minSize: 30 * 1024,
      maxSize: maxAssetSize
    }
  };
```

You can add this to the storybook webpack config.

**Reply by TowhidKashem** [reply]
To anyone copying the above snippet, the `minSize` key appears twice in it but otherwise it works in splitting up the chunks further for both `main.iframe.bundle.js` and `main.manager.bundle.js` and reducing the file size of any single bundle from going over the limit u set.

**Comment by bhagatali** [comment, marked as answer]
This thread in issue #4995 explains how to change webpack manager config.
